sklearn/ensemble/forest.py

```diff
@@ -69,6 +69,22 @@ class calls the ``fit`` method of each sub-estimator on random samples
 
 MAX_INT = np.iinfo(np.int32).max
 
+def _generate_sample_indices(random_state, n_samples):
+    """Private function used to _parallel_build_trees function."""
+    random_instance = check_random_state(random_state)
+    sample_indices = random_instance.randint(0, n_samples, n_samples)
+
+    return sample_indices
+
+def _generate_unsampled_indices(random_state, n_samples):
+    """Private function used to forest._set_oob_score fuction."""
+    sample_indices = _generate_sample_indices(random_state, n_samples)
+    sample_counts = bincount(sample_indices, minlength=n_samples)
+    unsampled_mask = sample_counts == 0
+    indices_range = np.arange(n_samples)
+    unsampled_indices = indices_range[unsampled_mask]
+
+    return unsampled_indices
```

Comment:
Why do we need these two functions? These are used only once. Hence I dont understand why we need to factor them out.